src/System.ComponentModel.Annotations/tests/MaxLengthAttributeTests.cs

```diff
@@ -20,25 +20,53 @@ protected override IEnumerable<TestCase> ValidValues()
             yield return new TestCase(new MaxLengthAttribute(-1), new int[20]);
             yield return new TestCase(new MaxLengthAttribute(15), new string[14]);
             yield return new TestCase(new MaxLengthAttribute(16), new string[16]);
-            yield return new TestCase(new MaxLengthAttribute(-1), new Collection<int>(new int[20]));
-            yield return new TestCase(new MaxLengthAttribute(15), new Collection<string>(new string[14]));
-            yield return new TestCase(new MaxLengthAttribute(16), new Collection<string>(new string[16]));
-            yield return new TestCase(new MaxLengthAttribute(-1), new List<int>(new int[20]));
-            yield return new TestCase(new MaxLengthAttribute(15), new List<string>(new string[14]));
-            yield return new TestCase(new MaxLengthAttribute(16), new List<string>(new string[16]));
 
             yield return new TestCase(new MaxLengthAttribute(16), new int[4, 4]);
             yield return new TestCase(new MaxLengthAttribute(16), new string[3, 4]);
+
+            // MaxLengthAttribute in the full .NET framework doesn't support ICollection.Count.
+            // See https://github.com/dotnet/corefx/pull/2650".
+            if (!PlatformDetection.IsFullFramework)
+            {
```

Comment:
This makes it a little bit confusing and when this gets fixed in Desktop then we would not be testing that and would be "hard" to find why we are not testing it. It would be better to have 2 tests one for netcore and one for netfx where each one receives its supported input and add SkipOnTargetFramework with the reason why it is skipped.